### PR TITLE
Feat: convert nodata to alpha TDE-625

### PR DESCRIPTION
**Pull request by MDavidson17**
[Example Output](https://basemaps.linz.govt.nz/?config=o215MBGedmy6nTiWwUgszffSJoTuGcsky3WTxPQ81qG4UCG47bAeC3Tx4Z9vZ8vfKWYGtSCJ1y7H2Fvbqk6GyRyK4U2zFKrizpfGRsdD5hDsnbUJXKMXZ2gpgfw3NUJ1F9MT1cvtaZYpyhgtCwQ6CVtmd6UTcYLGTNJjXvJ1Nmp2dE5WD7nmomoXnK7fAqJnzHqVL9u8JFS9etczhdhHbU9v1&i=test-chch&tileMatrix=NZTM2000Quad&debug#@-43.531450,172.631694,z9)

[Zoomed to 'Glitch' Example](https://basemaps.linz.govt.nz/?config=o215MBGedmy6nTiWwUgszffSJoTuGcsky3WTxPQ81qG4UCG47bAeC3Tx4Z9vZ8vfKWYGtSCJ1y7H2Fvbqk6GyRyK4U2zFKrizpfGRsdD5hDsnbUJXKMXZ2gpgfw3NUJ1F9MT1cvtaZYpyhgtCwQ6CVtmd6UTcYLGTNJjXvJ1Nmp2dE5WD7nmomoXnK7fAqJnzHqVL9u8JFS9etczhdhHbU9v1&i=test-chch&p=NZTM2000Quad&debug=true#@-43.5394150,172.6123336,z17.3381) - see https://toitutewhenua.atlassian.net/browse/BM-772

### CODE REVIEW

File: scripts/files/file_tiff.py [hunk 1]
```diff
@@ -96,22 +122,34 @@ def check_no_data(self, gdalinfo: GdalInfo) -> None:
...
                 )
         else:
-            self.add_error(error_type=FileTiffErrorType.NO_DATA, error_message="noDataValue not set")
+            if len(bands) == 4 and bands[3]["colorInterpretation"] != "Alpha":
+                self.add_error(error_type=FileTiffErrorType.NO_DATA, error_message="noDataValue not set")
+
+    def is_no_data(self, gdalinfo: GdalInfo) -> bool:
+        """return True if there is a "noDataValue" and it is set to 255 in the "bands"."""
+        bands = gdalinfo["bands"]
+        if "noDataValue" not in bands[0]:
```
Comment (paulfouquet) [inline scripts/files/file_tiff.py:131]: This is more clear to me and also match with the function comment:
```
if "noDataValue" in bands[0] and bands[0]["noDataValue"] == 255:
       return True
return False
```

File: scripts/files/file_tiff.py [hunk 2]
```diff
@@ -44,7 +44,9 @@ def set_scale(self, scale: int) -> None:
     def set_path_standardised(self, path: str) -> None:
         self._path_standardised = path
 
-    def get_gdalinfo(self) -> Optional[GdalInfo]:
+    def get_gdalinfo(self, original: Optional[bool] = False) -> Optional[GdalInfo]:
```
Comment (paulfouquet) [inline scripts/files/file_tiff.py:47]: In that case I won't bother handling `original` in that function but make `get_gdalinfo_original` accessible and call it instead.

File: scripts/files/file_tiff.py [hunk 3]
```diff
@@ -96,22 +122,34 @@ def check_no_data(self, gdalinfo: GdalInfo) -> None:
                     custom_fields={"current": f"{current_nodata_val}"},
                 )
         else:
-            self.add_error(error_type=FileTiffErrorType.NO_DATA, error_message="noDataValue not set")
+            if len(bands) == 4 and bands[3]["colorInterpretation"] != "Alpha":
+                self.add_error(error_type=FileTiffErrorType.NO_DATA, error_message="noDataValue not set")
+
+    def is_no_data(self, gdalinfo: GdalInfo) -> bool:
```
Comment (paulfouquet) [inline scripts/files/file_tiff.py:128]: You can make `gdalinfo` as optional (for testing?) and call `get_gdalinfo_original()` if `_gdalinfo_original` is None

File: scripts/standardising.py [hunk 4]
```diff
@@ -100,12 +101,17 @@ def standardising(file: str, preset: str, cutline: Optional[str]) -> FileTiff:
             run_gdal(get_cutline_command(input_cutline_path), input_file=input_file, output_file=target_vrt)
             input_file = target_vrt
 
+        gdalinfo_original_tiff = tiff.get_gdalinfo(original=True)
+        if not cutline and gdalinfo_original_tiff and tiff.is_no_data(gdalinfo_original_tiff):
```
Comment (paulfouquet) [inline scripts/standardising.py:105]: `elif gdalinfo_original_tiff and tiff.is_no_data(gdalinfo_original_tiff):`?

File: scripts/files/file_tiff.py [hunk 5]
```diff
@@ -96,22 +122,34 @@ def check_no_data(self, gdalinfo: GdalInfo) -> None:
...
+                bands_num = 4
+        if len(bands) != bands_num:
             self.add_error(
                 error_type=FileTiffErrorType.BANDS,
-                error_message="bands count is not 3",
+                error_message=f"bands count is not {bands_num}",
                 custom_fields={"count": f"{int(bands_num)}"},
             )
 
     def check_srs(self, gdalsrsinfo_tif: bytes) -> None:
```
Comment (paulfouquet) [inline scripts/files/file_tiff.py:151]: Unit test missing

File: scripts/files/file_tiff.py [hunk 6]
```diff
@@ -96,22 +122,34 @@ def check_no_data(self, gdalinfo: GdalInfo) -> None:
...
+    def is_no_data(self, gdalinfo: GdalInfo) -> bool:
+        """return True if there is a "noDataValue" and it is set to 255 in the "bands"."""
+        bands = gdalinfo["bands"]
+        if "noDataValue" not in bands[0]:
+            return False
+        if bands[0]["noDataValue"] != 255:
+            return False
+        return True
 
     def check_band_count(self, gdalinfo: GdalInfo) -> None:
```
Comment (paulfouquet) [inline scripts/files/file_tiff.py:137]: Unit test missing

File: scripts/files/file_tiff.py (hunk 3 above)
Comment (paulfouquet) [inline scripts/files/file_tiff.py:128]: Unit test missing

**Review by paulfouquet — DISMISSED** [review]

**Review by paulfouquet — DISMISSED** [review]

**Review by amfage — APPROVED** [review]

**Review by amfage — APPROVED** [review]

**Pull request merged**